src/topology/algebra/module/locally_convex.lean

```diff
@@ -0,0 +1,138 @@
+/-
+Copyright (c) 2022 Anatole Dedecker. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Anatole Dedecker
+-/
+import analysis.convex.topology
+import analysis.seminorm
+/-!
+# Locally convex topological modules
+
+A `locally_convex_space` is a topological semimodule over an ordered semiring in which any point
+admits a neighborhood basis made of convex sets, or equivalently, in which convex neighborhoods of
+a point form a neighborhood basis at that point.
+
+In a module, this is equivalent to `0` satisfying such properties.
+
+## Main results
+
+- `locally_convex_space_iff_zero` : in a module, local convexity at zero gives
+  local convexity everywhere
+- `seminorm.locally_convex_space` : a topology generated by a family of seminorms is locally convex
+- `normed_space.locally_convex_space` : a normed space is locally convex
+
+## TODO
+
+- define a structure `locally_convex_filter_basis`, extending `module_filter_basis`, for filter
+  bases generating a locally convex topology
+- show that any locally convex topology is generated by a family of seminorms
+
+-/
+
+open topological_space filter
+
+open_locale topological_space
+
+section semimodule
+
+/-- A `locally_convex_space` is a topological semimodule over an ordered semiring in which convex
+neighborhoods of a point form a neighborhood basis at that point. -/
+class locally_convex_space (𝕂 E : Type*) [ordered_semiring 𝕂] [add_comm_monoid E] [module 𝕂 E]
```

Comment:
please use `𝕜` instead of `𝕂`.

Reply:
Looking at it again, you should probably rename it to `R` since it is not a normed ring.

Reply:
I've used `𝕜` in all convexity files. The point of using unicode characters for types is that they're easily identifiable and can't be accidentally grepped.